CMakeLists.txt

```diff
@@ -166,6 +166,7 @@ if (WIN32)
     # "VCRUNTIME140.dll not found. Try reinstalling the app.", but give users
     # a choice to opt for the shared runtime if they want.
     option(USE_STATIC_CRT "Link against the static runtime libraries." ON)
+    set(CMAKE_CXX_STANDARD 20)
```

Comment:
What doesn't compile with C++17 for you?

Reply:
I get error `C7555` in [DriverEnums.h line 1109 and 1113](https://github.com/google/filament/blob/45f5a07989861b9b01782c4a8ba188a6ea107f55/filament/backend/include/backend/DriverEnums.h#L1109)

`use of designated initializers requires at least '/std:c++20'`

This also happends in other files like [here](https://github.com/google/filament/blob/45f5a07989861b9b01782c4a8ba188a6ea107f55/filament/include/filament/Box.h#L180) or [here](https://github.com/google/filament/blob/45f5a07989861b9b01782c4a8ba188a6ea107f55/libs/viewer/src/Settings.cpp#L587)

Reply:
We can fix these two lines in `DriverEnums.h` easily so the don't use designated initializers.

Regardless, I thing this is a problem is other places and I thought we were compiling with C++20 already on windows, specifically because of that?

Reply:
My previous comment only meant that we shouldn't use C++20 globally across the project, since 1P clients are limited to C++1 7 at the moment. On Windows we have more flexibility.

On Windows, we're currently using `/std:c++latest` for MSVC, but maybe we need to explicitly use `/std:c++20` (apparently those two flags have slightly different meanings). @dabeschte can you try replacing this line with `/std:c++20` and seeing if it fixes the designated initializer error?

https://github.com/google/filament/blob/626577fe3d3697d3a5f3cca691b24e0e57f159aa/CMakeLists.txt#L285

Reply:
No that does not work, since the if (MSVC) 3 lines below overrides that. This results in MSVC using the latest preview, not the latest stable compiler version.
![image](https://github.com/google/filament/assets/11183241/6416e463-3969-4fd7-8478-eea358713a60)

The error that I get in this case is 
```
Error	C3878	syntax error: unexpected token 'this' following 'expression'	backend	<filament>\libs\utils\include\utils\Range.h	37	
```

Caused by this piece of code that actually looks fine to me, but since we are using preview features, I assume the compiler might have a bug here?

```
    bool overlaps(const Range<T>& that) const noexcept {
        return that.first < this->last && that.last > this->first;
    }
```

Anyways, changing line 289 in CMakeLists.txt to "/std:c++20" solves the problem for Windows.
That said, MSVC is actually right about the invalid usage of C++20 features in multiple files - not only in DriverEnums.h 
[G++ just doesn't show a warning by default](https://stackoverflow.com/questions/69009389/how-to-get-away-with-using-designated-initializers-in-c17-or-why-is-it-seemi#:~:text=understanding%20is%20backward.-,C%2B%2B17%20does%20not%20support%20designated%20initialisers%20at%20all,acceptance%20in%20the%20standardisation%20process.)

Reply:
`C3878` has just been fixed.